frontend/src/components/common/Resource/EditorDialog.tsx

```diff
@@ -152,45 +164,31 @@ export default function EditorDialog(props: EditorDialogProps) {
     );
 
     setCode({ code: value as string, format: code.format });
+    if (error && getObjectsFromCode({ code: value as string, format: originalCode.format })) {
+      setError('');
+    }
 
     if (onEditorChanged) {
       onEditorChanged(value as string);
     }
   }
 
-  function getObjectFromCode(codeInfo: typeof originalCode) {
-    function looksLikeJson(code: string) {
-      const trimmedCode = code.trimRight();
-      const lastChar = !!trimmedCode ? trimmedCode[trimmedCode.length - 1] : '';
-      const firstChar = !!trimmedCode ? trimmedCode[0] : '';
-      if (['{', '['].includes(firstChar) || ['}', ']'].includes(lastChar)) {
-        return true;
-      }
-
-      return false;
-    }
-
+  function getObjectsFromCode(codeInfo: typeof originalCode): {
+    obj: KubeObjectInterface[] | null;
+    format: string;
+    error: Error | null;
+  } {
     const { code, format } = codeInfo;
-    const res: { obj: KubeObjectInterface | null; format: string; error: Error | null } = {
+    const res: { obj: KubeObjectInterface[] | null; format: string; error: Error | null } = {
       obj: null,
       format,
       error: null,
     };
 
-    if (!res.obj) {
-      res.format = 'yaml';
-      try {
-        res.obj = yaml.load(code) as KubeObjectInterface;
-        return res;
-      } catch (e) {
-        res.error = new Error((e as Error).message || t('Invalid YAML'));
-      }
-    }
-
     if (!format || (!res.obj && looksLikeJson(code))) {
       res.format = 'json';
       try {
-        res.obj = JSON.parse(code) as KubeObjectInterface;
+        res.obj = JSON.parse(code) as KubeObjectInterface[];
```

Comment:
The `as` change here doesn't convert the value of the obj, only forces it to evaluate as that type (this is why it doesn't complain even when you parse a single JSON object).
As a result, you can try a JSON single object and it's currently not possible to handle it.

A fix for this is to first parse the code and check if it's an array, if not, add it as an array.

Reply:
fixed